### DISCUSSION TITLE
Additions to LinkStack

### DISCUSSION THREAD

**Discussion by JulianPrieber** [JulianPrieber/littlelink-custom#49] [In progress]
# Additions to LinkStack
#### Table of contents:
- [Minor additions](#1)
- [More substantial additions](#2)
- [Major additions](#3)
- [Canceled additions](#4)

<br>

<a name="labels"></a>
#### Labels:
- 🆘 ➤ Help wanted (unless marked as done)
- ❓ ➤ Feedback needed (unless marked as done)
- 🚧 ➤ Still needs some work (even if marked as done)

<br>

<a name="1"></a>
#### Any minor addition that should be implemented.
Minor additions are features that can mostly be implemented by editing already existing code.

**[`^        back to top        ^`](#)**

<br>

- [x] Allowed admins to change their email regardless if email auth is enabled or not
- [x] Changed title to name on littlelink page
- [x] Limit littlelink description width
- [x] Add spaces and headings
- [x] Add gradient to plain background colors
- [x] Add option to switch between light and dark mode
- [x] Don't force centering text on footer pages
- [x] Increase profile picture size on littlelink pages (possible with themes)
- [x] Hide links table when no links have been added (https://github.com/JulianPrieber/littlelink-custom/issues/172)

<br>

<a name="2"></a>
#### More substantial addition that should be implemented.
More substantial additions are features that require a moderate number of lines of code added, with not much time needed to implement such solutions.

**[`^        back to top        ^`](#)**

<br>

- [x] Make button editor optional
- [x] Optimize all pages for mobile view [🚧](#labels) [🆘](#labels)
- [x] Allow for HTML syntax on littlelink page description
- [x] Add beta-version support to updater
- [x] Google Analytics support?[❓](#labels)
- [x] Make share button optional for users[❓](#labels)
- [x] Add ability to delete Themes
- [x] Add ability to delete users
- [x] Add option to verify user's E-mails manually
- [x] Add option for admins to add users from the Admin Panel
- [x] Enable custom JavaScript for Themes
- [x] Allow for custom button icons in Themes
- [x] Add option in themes to disable custom buttons
- [x] Add ability to open links in same tab
- [x] Add option to theme config to use the default buttons
- [x] Add subtitle text block
- [x] Enable changing texts of footer links and titles
- [x] Tool for sending test emails
- [x] Improve security for profile URLs without prefix
- [ ] Copy to clipboard button
- [ ] Gradient angle slider for button editor [🆘](#labels)
- [ ] Add ability to change default URL prefix
- [x] Add option to disable click count and open links directly
- [ ] Use [Fork Awesome](https://forkaweso.me/Fork-Awesome/) instead of Font Awesome

<br>

<a name="3"></a>
#### Major additions that should be implemented.
Major additions are features that require numerous lines of code, a new page or extensive time to implement.

**[`^        back to top        ^`](#)**

<br>

- [x] Add Button editor
- [x] Custom Theme support
- [x] Automatic updater
- [x] Add ability to arrange links by 'dragging and dropping' ([SortableJS](https://github.com/SortableJS/Sortable)) [🆘](#labels)
- [x] Add theme updater
- [x] Add ability for user to delete their account
- [x] Expand and improve security check tool
- [x] Add option to export buttons/links
- [ ] Ability to embed (in external site) https://github.com/JulianPrieber/littlelink-custom/discussions/121
- [ ] Add some form of brute force prevention
- [ ] Automatically detect pre-defined site from URL
- [ ] Add video embed block

<br>

<a name="4"></a>
#### Canceled additions that probably won't be implemented.
Additions that can't be realized without rewriting a lot of the application. Help on these additions is always welcome.

**[`^        back to top        ^`](#)**

<br>

- [x] Translation into other languages

<br>

#### Feel free to submit additions below.

**Comment by Doncellup** [comment]
Hi 👋 (we talked about it on Discord)

To solve the security problem of when the @ prefix is ​​removed, it would be necessary to blacklist all the routes already used (those which would cause problems)

This issue may seem minor, but actually removing the prefix (by default @) can really be a great addition to the project!!

Thanks in advance,
Doncellup

**Reply by JulianPrieber** [reply]
I'm probably not gonna remove the @ by default.

I might add a system that allows changing the default URL prefix in the next update of the Advanced Config.

As for the blacklist, I would have to look into how to realize this. As for now, this is this something that is too complex for me to solve for a problem that only affects a tiny minority of people.
This issue is pretty low on my list and might take a while to get implemented.

**Reply by JulianPrieber** [reply]
@Doncellup 

I worked out the blacklist system, and you can fully use this feature now. If enabled in your config, you can use URLs without the @ prefix.
I might include a full implementation to use this as the default action in the next advanced config update.

You can use this feature if you install the latest beta version.

**Reply by Doncellup** [reply]
> @Doncellup
> 
> I worked out the blacklist system, and you can fully use this feature now. If enabled in your config, you can use URLs without the @ prefix. I might include a full implementation to use this as the default action in the next advanced config update.
> 
> You can use this feature if you install the latest beta version.

Awesome !!  I'll try that later, I don't have time right now.  Thank you very much for implementing this !

**Reply by arcanation** [reply]
hello, is there any tutorial how to remove @ prefix, I try to go to advanced config but there's no option to remove @ prefix,
is there anyway to do it?

**Comment by Doncellup** [comment]
Another suggestion:

Currently, to change the order of the buttons, you have to number them all one by one, which is long.  The worst part is that as soon as you want to add one, you have to renumber everything.

To remedy this problem, there would have to be a drag and drop system like this:
![7E6146A4-064D-49A9-A3E3-391F1CF3BFC2](https://user-images.githubusercontent.com/90469321/179600418-1b203d82-2aa4-4018-b415-933c1a248632.jpeg)

Thanks in advance,
Doncellup

**Reply by JulianPrieber** [reply]
Awesome addition, I would love to see this implemented.

I will put this on the list and mark this as help wanted. I'm not really comfortable working with JavaScript, meaning I won't be able to add this any time soon.

If anyone would like to get this working, I would be happy to help where I can.

**Reply by JulianPrieber** [reply]
@Doncellup 

We actually managed to work out a solution that implements this. We're doing a public beta test on: https://demo.littlelink-custom.com

**Comment by Doncellup** [comment]
Hello ! One more suggestion! I find that currently the share button is a bit vulgar, it takes up as much space as the link buttons…
I think it should be replaced by a small logo like this (it comes from Linktree):
![image](https://user-images.githubusercontent.com/90469321/181371367-fd0dc59c-b62e-42dd-a2d6-d03303a1a52e.png)

**Comment by Doncellup** [comment]
One more suggestion!!

On PC (large screen) the buttons seem really small (they are the width of a phone screen…).
![image](https://user-images.githubusercontent.com/90469321/181372374-ae44635d-2c26-4a05-92e9-8fed24bf6ba9.png)

It would be really nice to widen them more (like on Linktree) and that they resize according to the width of the screen, with a maximum width anyway.
![image](https://user-images.githubusercontent.com/90469321/181372479-7fc5db3b-f3df-48b2-b6b3-753453e85d90.png)

It would be nice to do that natively, but normally it's the themes that handle that?

**Reply by JulianPrieber** [reply]
Again, as stated above, we use the frontend provided by the project LittleLink and will probably continue to use mostly use that unchanged as the default option.

However, if you visit your link page on mobile or resize your screen to a specific size, you can observe that the share button resizes to fit the smaller screen size better.

But yes, **everything can be adjusted however you'd like with themes**. This includes the normal/share button sizes and scaling of all elements on multiple resolutions.

**Reply by Doncellup** [reply]
Hmm okay, I will inquire to create my personalized theme

**Comment by MagicLike** [comment]
What about an updater for themes?
For example, when a theme is added, a pull/git link is included so that LLC can automatically detect when something changes or a new version is released. Then the user gets notified and LLC can update the theme by itself.

**Reply by JulianPrieber** [reply]
I have thought about something like this. 
I probably won't implement a full-blown updater, but I might add something like an update notification.

When first implementing the theme system, I wanted to have something that would download themes directly from the UI on your personal instance with one click, similar to word press themes. But I scrapped that idea for the sake of simplicity.
I might add a specked down version of this later on.

My main problem is deleting themes after an update because I cannot expect themes to have the original folder names when downloaded. I would have to figure something out using the readme.md and the version number in there, but even I forget to update that one after updating from time to time.

For this reason, I'm still not sure if that's a good idea.

**Reply by JulianPrieber** [reply]
I had to rework parts of the system that uploads themes, and I ended up removing version number from theme folders.

I still want some form of information which version of a theme is installed, so I ended up working out a prototype for a theme updater:
![chrome_I7481UFu5a](https://user-images.githubusercontent.com/60265788/189665835-c71e7892-7286-4cbd-bfdc-3631b9d7ebb1.png)

I could get this thing working if I could somehow get a direct link to the GitHub release. My problem is that there is no such thing in themes currently.

What I could do is use the creator name for the GitHub user and theme name for the repository. That would work, but only because all themes on GitHub are made by me...
So I can't expect this to work in the future.

My best idea is to make an update for every theme I made including a GitHub link and also including that in the template file.
This way, everyone had to update every theme they have manually one last time, but then the updater could kick in.

https://github.com/JulianPrieber/llc-default/commit/167b493232a15b617613a7a9d2db04b16609a443

**Reply by MagicLike** [reply]
Sound like a very nice idea! 👀

**Reply by JulianPrieber** [reply]
So I'm stuck at this one step for quite a while now.
I need to find something to test if a certain GitHub API page exists, but I just can't find anything that works.
I tried so many solutions, but I can find anything that works with the GitHub API page. I'm not sure why that is. Most websites like llc.bio work just fine, but api.github.com seems to block certain requests.

This is the last thing I tried:

```
<?php
  function urlExists($url=NULL)
  {
      if($url == NULL) return false;
      $ch = curl_init($url);
      curl_setopt($ch, CURLOPT_TIMEOUT, 5);
      curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 5);
      curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
      $data = curl_exec($ch);
      $httpcode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
      curl_close($ch); 
      if($httpcode>=200 && $httpcode<300){
          echo "true";
      } else {
          echo "false";
      }
  }
  
urlExists('https://api.github.com/repos/JulianPrieber2/galaxy/releases/latest')
?>
```

**Reply by JulianPrieber** [reply]
This is what it would look like if it worked:

![chrome_n0aRZmM1kA](https://user-images.githubusercontent.com/60265788/189726828-750239bd-91d5-4702-80c1-fd341af7843c.png)

I still can't get the request to api.github.com to work, so I currently just check if the GitHub repository exists. This will still trow an error if the uploaded theme does not have a release.

Due to all the http requests, the page takes quite a while now to load. I think this is tolerable to a degree, but the request waiting for the 404 error from the theme Aurora results in the page taking a couple seconds to load.

Aurora does not have a GitHub repository, and I need a way to check for that. I hope I can find a faster way of doing this, otherwise I might have to scrap this feature.

**Reply by JulianPrieber** [reply]
I got it working and It's definitely faster now!

**Reply by MagicLike** [reply]
Sorry, I couldn't aswer. But I am glad, that it worked!
Maybe you could create a repo for the Aurora theme - with permission of the Discord user - and add the credit there?

**Reply by JulianPrieber** [reply]
That's probably not a bad idea.

This is the author: https://me.unraid.nz/theme/@kiwi

**Reply by JulianPrieber** [reply]
**I got the theme updater working!**
There are still a few things to iron out, but It's pretty much fully functional.

I have the latest version of the thing on the beta server.
I still have to add the source code section to the themes, so most themes display as "Update manually" on the updater.

You can test the updater with this theme:
[galaxy-1.0-test.zip](https://github.com/JulianPrieber/littlelink-custom/files/9556932/galaxy-1.0-test.zip)


The biggest problem currently is that the GitHub API request aren't authenticated, so the rate limit is at 100 request per hour or something like that.
With a request made for every theme every time the page gets refreshed, this limits gets reached rather quickly.
Other than this, the updater works okay as it stands right now.

**Reply by JulianPrieber** [reply]
I made a theme file with all the themes that you can just upload as you normally would.
[themes.zip](https://github.com/JulianPrieber/littlelink-custom/files/9557826/themes.zip)

This includes all the updated themes with the new source code section added.

**Reply by JulianPrieber** [reply]
Everything seems to be working :)

![chrome_aWFsn5DCXi](https://user-images.githubusercontent.com/60265788/189922535-45e9ee30-87a7-4609-b3fd-6a9c29ae31ad.png)

**Reply by MagicLike** [reply]
I am just setting up my beta install - I will test it then...

**Reply by MagicLike** [reply]
For the problems with the GitHub API:
We could make the auto-updater optional and include it in the initial setup / config.
Everyone, who wants to use it, needs to provide their own API key (I think thise are limited with 5000 request/h) and therefore a GitHub account.

**Reply by MagicLike** [reply]
Tested it, runs nice - but apparently the rate limit is already reached...

**Reply by MagicLike** [reply]
> Sorry, I couldn't aswer. But I am glad, that it worked! Maybe you could create a repo for the Aurora theme - with permission of the Discord user - and add the credit there?

Kiwi doesn't have any problems with a repo created on his/her behalf. But I think #160 would be better...

**Reply by MagicLike** [reply]
One small addition: The table should show the newest version available version number next to the installed one.

**Comment by ugintl** [comment]
**Allow admin to customize user permissions**

As an admin, I want to be able to create roles and customize user permissions. Is it something in your to do list?

**Reply by JulianPrieber** [reply]
Probably not any time soon.

To what extent would you like to customize permission exactly?

**Reply by ugintl** [reply]
crud permissions. What we normally see in laravel apps. Ability to create roles and limit their access using permissions. In LLC I see there are already roles like admin, vip and normal user. Let me know if you need more clarification. Which laravel package is LLC using for auth and user management?

**Reply by JulianPrieber** [reply]
I didn't write this system, so I'm not sure.
I don't even know what the VIP group does. I know it does something, but I forgot what exactly.

**Comment by MagicLike** [comment]
"Update manually" should link to the Theme repo (if linked)

**Reply by JulianPrieber** [reply]
Unlikely that I will add this. The system uses this when no link to any repo is present, so there is nothing I could link to.

I could link it to the theme page.

**Reply by MagicLike** [reply]
Yeah, I did not think of that...
But that is a better idea!

**Reply by JulianPrieber** [reply]
https://github.com/JulianPrieber/littlelink-custom/commit/8e232344d7d971740352fd796847475ae9105bf6

**Comment by MagicLike** [comment]
"Update all themes" icon should spin, when pressed

**Reply by JulianPrieber** [reply]
I have no idea how to do that xD

**Reply by JulianPrieber** [reply]
So uhm.. apparently, I do know how to do this.

https://user-images.githubusercontent.com/60265788/190174416-51afdff8-92e2-405f-8cd2-884a7ea493a9.mp4



I just can't seem to figure out how to make the icon spin without it shifting :face_with_spiral_eyes:

**Reply by MagicLike** [reply]
Got it working here: https://codepen.io/MagicLike/pen/wvjoyyr
(without shifting)

**Reply by JulianPrieber** [reply]
https://codepen.io/julianprieber/pen/gOzLeGK

**Reply by JulianPrieber** [reply]
No shift =
![chrome_tdbYTmI9ZU](https://user-images.githubusercontent.com/60265788/190177493-7626d463-5ba9-4241-87fa-961b8b3e5e31.png)

**Reply by JulianPrieber** [reply]
Is there a CSS selector for click events?

**Reply by MagicLike** [reply]
OK, am I right, that it spins without shift on codepen, but with shift in LLC?

**Reply by JulianPrieber** [reply]
You can see the shift clearly on codepen

**Reply by JulianPrieber** [reply]
With shift I mean that the icon moved to ther top of the button

**Reply by MagicLike** [reply]
> Is there a CSS selector for click events?

I think only when the mouse button is clicked...
https://stackoverflow.com/questions/23550000/css-click-events#23550026

**Reply by MagicLike** [reply]
> You can see the shift clearly on codepen

On my side ther is no shift...
EDIT: there is, but it is so minimal that it is hard to tell...

**Reply by JulianPrieber** [reply]
> https://stackoverflow.com/questions/23550000/css-click-events#23550026

They cheat using a checkbox.

**Reply by JulianPrieber** [reply]
> On my side ther is no shift...

Wait what?

**Reply by MagicLike** [reply]
There actually is - it's minimal

**Reply by JulianPrieber** [reply]
What do you mean by minimal?

The icon is either above or next to the text.

This:
![chrome_FO36bmAX3S](https://user-images.githubusercontent.com/60265788/190179880-65b21bfc-453d-4ffb-bbaf-21120e2a366e.png)

vs

![chrome_tdbYTmI9ZU](https://user-images.githubusercontent.com/60265788/190179918-37eb76eb-9fad-4cdd-902b-dec2a76fea4f.png)

**Reply by MagicLike** [reply]
> You can see the shift clearly on codepen

You need to add the following CSS to stop the wobbling:
```
transform-origin: 50% 50%;
display: inline-block;
```

**Reply by JulianPrieber** [reply]
Yess I actually just implemented that xD

**Reply by JulianPrieber** [reply]
https://github.com/JulianPrieber/littlelink-custom/commit/fe78560368e6c8e18516c303274b227cfd33d3fe

**Reply by MagicLike** [reply]
> What do you mean by minimal?
> 
> The icon is either above or next to the text.
> 
> This: ![chrome_FO36bmAX3S](https://user-images.githubusercontent.com/60265788/190179880-65b21bfc-453d-4ffb-bbaf-21120e2a366e.png)
> 
> vs
> 
> ![chrome_tdbYTmI9ZU](https://user-images.githubusercontent.com/60265788/190179918-37eb76eb-9fad-4cdd-902b-dec2a76fea4f.png)

Ahhh, we talked at cross purposes...
I meant, that the icon was wobbling....

**Reply by JulianPrieber** [reply]
Yeah :woozy_face:

**Reply by MagicLike** [reply]
The icon is still wobbling...

**Reply by JulianPrieber** [reply]
Yeah, the Bootstrap Icon isn't centered right I think.

**Reply by JulianPrieber** [reply]
I'm not sure if this is better: https://codepen.io/julianprieber/pen/gOzLeGK

**Reply by MagicLike** [reply]
No, nothing changed...
But I think this CSS line should fix the wobbling:
`transform-origin: 50% 50%;`

**Reply by JulianPrieber** [reply]
Unfortunately, that doesn't do it.

**Reply by MagicLike** [reply]
But I think it's okay for now because it's barely noticeable.

**Comment by woutr-nl** [comment]
A thing that i'm missing right now is the ability to test if the mail is working.

**Reply by JulianPrieber** [reply]
Added to the list 👍

**Comment by JOY** [comment]
Hi. I'd like to suggest the verification feature in which user can authorize their social accounts to have a green check. 
Ex: In Unstoppable domain, I have to login and authorize the social accounts to get them listed: https://ud.me/overmint.x

**Comment by ADHDefy** [comment]
Hi! I would like to see Calckey added as a predefined site. It is a rapidly growing, feature-rich Fediverse client based on Misskey.

https://calckey.org/

Thank you! :)

EDIT: I would still like to see Calckey added, and I'd also like to request kbin. It is yet another new and bustling Fediverse client. It's a link aggregator, like Lemmy. You can see more here: https://kbin.pub/en

**Comment by UberMetroid** [comment]
It is a big ask, but I would like LDAP added for user control. Thanks!

**Comment by spikehidden** [comment]
Since Linkstack has multi domain support why not make it possible to show a different default site on the homepage based on the domain or load a .env file per domain.

Right now I either have to completely install and Setup LinkStack a second time or try making a redirect via Cloudflare.

**Reply by adhodgson1** [reply]
Have you got this multi-domain redirect working currently? I've tried a few recipes using Nginx but have fallen short.

**Reply by JulianPrieber** [reply]
```
  if ($host = 'example.com') {
    return 301 https://xyz.com$request_uri;
  }
```

**Reply by zahornyak** [reply]
Do you know if that has still not been implemented?

**Comment by h0ek** [comment]
2fa 👍🏻

**Comment by Eamourinho** [comment]
OAuth/OIDC/Social sign on? :D

**Reply by JulianPrieber** [reply]
Social login is already implemented

**Reply by D1reW0lf** [reply]
Do you know if there are any changes adding Mastodon to the list of supported sites?
I tried to add this plug-in without any luck.
(It crashed my site but nothing a backup couldn't handle. :D )

https://github.com/kawax/socialite-mastodon

Thank you.

**Reply by tylertufano** [reply]
Especially for the self-hosted crowd, add support for additional SSO login providers (Authelia, Authentik, Keycloak, etc.).

Socialite Providers provides support for different services according to their documentation (https://socialiteproviders.com/Authelia/). Or, using generic OIDC provider (https://github.com/Kovah/laravel-socialite-oidc).

Additionally, add the option to attempt to login automatically with the preferred provider.

**Comment by timscha** [comment]
I am missing a some Page Icons. I would love to see icons for Behance (that's missing in general) and Spotify.

Also it would be nice to change the footer links completely on my own (for example, add a link to an existing privacy page).

**Comment by soja92** [comment]
Would be useful to be able to manage multiple pages under a single account.

**Reply by forgeOneHQ** [reply]
That would be an awesome feature.

**Comment by narvaldreams** [comment]
Hey, a few days ago I paid the Sponsor 8 dlls and when I tried to configure the DNS it was not possible, I sent an email informing that situation and how the access would be for the administration of said process and they NEVER answered me.
Not even an email to let me check or any help, they just ignored me.
I hope that here you can answer me and give me the support I need.

**Reply by JulianPrieber** [reply]
Hello Miguel,

Thank you for reaching out and bringing this to our attention.

I sent an email with detailed instructions 8 days ago to the provided email through the form. I've just checked my spam folder and didn't receive any emails from you. Let's try again. You can contact support directly at info@linkstack.org for assistance with your DNS configuration issue.

Please note, this thread is primarily for feature requests, not support inquiries.

**Reply by JulianPrieber** [reply]
Hi @narvaldreams ,

Just checking in. did you manage to send an email to our support address? I haven't received anything yet. Please let me know if there are any other ways you'd like to connect.

Thanks,
Julian

**Comment by stock-reaper** [comment]
Suggestion: It'd be good to be able to insert images within the text editor fields for example inside the "text block" or inside the page description. Even the ability to embed images in text fields with <img> tags would be great!

**Reply by pa-philos** [reply]
I'd love that as well. Would allow for a more customizable web solution.

**Comment by PYU224** [comment]
Hello.
If you add linkstack crypto address, I want to create instance.
Would you add crypto address in additional links?

For Example, this cointree.
https://cointr.ee/

**Comment by wittywilla** [comment]
I would love for there to be the addition of the Threads social media site in the predetermined social media link templates - or the ability to set custom icons for custom links.

One more addition I'd love to see would be a block that you can write out titles & paragraphs and whatnot with proper formatting tools - as well as adding text size and formatting tools for the profile page description (like choosing the width in an easy to find manner)

**Comment by ZicPL** [comment]
ˆ
—————————————————> RECENTLY very popular : **_"[Links Page for account users only](https://github.com/orgs/LinkStackOrg/discussions/804#discussioncomment-11389710)"_** <—————————————————

º
¹ Maybe someone here will sponsor this feature and we will see it much faster ✌️
² It's worth reading the discussion because the Feature seems more interesting than the ideas above (a very practical feature).
³

**Comment by thealexparadox** [comment]
addition of fediverse:creator meta tag
it could go under 'custom meta tags' in the config

syntax:
'fediverse:creator' => '@your@fediverse.address', // fediverse @address

further info here (https://rknight.me/blog/highlighting-journalism-with-the-fediverse-creator-tag/)

**Comment by thealexparadox** [comment]
addition of fediverse:creator meta tag
it could go under 'custom meta tags' in the config

syntax:
'fediverse:creator' => '@you@your.server', // fediverse @address

further info here (https://rknight.me/blog/highlighting-journalism-with-the-fediverse-creator-tag/)

**Comment by arazilsongweaver** [comment]
Add an optional basic email form to the "contact" page. This would allow users to receive emails from the site without publishing that email address on the site. This feature should also include a CAPTCHA and other anti-spam features to prevent bot abuse.

**Comment by iBasalamah** [comment]
Would you mind adding a button to rest the analytics?

**Comment by mperson91** [comment]
Any plans for simple form submission to collect an email list or a poll? Soundcloud box to auto play a song? Just a few ideas. Great project!!

**Comment by mdn-united** [comment]
A visitor counter on the page or a counter that shows how many page impressions there have been. Not everyone who opens a page clicks on a link.

Ein Besucherzähler auf der Seite oder ein Zähler, der anzeigt, wie viele Seitenaufrufe es gab. Nicht jeder, der eine Seite öffnet klickt auf einen Link.

**Comment by scuzy98** [comment]
Any chance for a picture block to add additional pictures?

**Comment by JulianPrieber** [comment]
https://github.com/LinkStackOrg/LinkStack/discussions/978
